kuma/static/js/wiki-edit-draft.js

```diff
@@ -0,0 +1,428 @@
+            text += draftWarning;
+            text += '<button type="button" class="js-restoreLink btn link">' + restore + '</button> ';
+            text += '<button type="button" class="js-discardLink btn link">'+ discard +'</button>';
+            var $contentNode = $('#id_content');
+            var editor;
+
+            // Notify user of the existing draft
+            $draftOld.empty();
+            $draftOld.append(text);
+
+            // Hook up the "restore" link`
+            $draftDiv.find('.js-restoreLink').on('click', function() {
+                if(draftRevision) {
+                    // can't use ID because form has duplicate IDs #facepalm
+                    $revisionInputs.val(draftRevision);
+                }
+                $contentNode.val(prevDraft);
+                editor = $contentNode.ckeditorGet();
+                editor.setData(prevDraft);
+                editor.focus();
+                // inform user and remove old draft element
+                $draftOld.text(gettext('Draft restored.')).delay(3000).slideUp();
+                saveDraft();
+
+                mdn.analytics.trackEvent({
+                    category: 'draft',
+                    action: 'restore'
+                });
+            });
+
+            // Hook up the "discard" link
+            $draftDiv.find('.js-discardLink').on('click', function() {
+                var currentDraftTime = localStorage.getItem(DRAFT_NAME + '#save-time');
+                // only clear localstorage if no new draft created
+                if(draftTime === currentDraftTime) {
+                    clearDraft();
+                }
+                // inform user and remove old draft element
+                $draftOld.text(gettext('Draft discarded.')).delay(3000).slideUp();
+
+                mdn.analytics.trackEvent({
+                    category: 'draft',
+                    action: 'discard'
+                });
+
+            });
+        }
+    }
+
+
+    // Save button submits to top-level
+    $form.on('submit', function () {
+        if (supportsLocalStorage) {
+            enableAutoSave(false);
+            clearTimeout(draftTimeOutId);
+        }
+        return true;
+    });
+
+
+    // update the UI to reflect status of draft
+    function updateDraftState(action) {
+        var now = new Date();
+        var nowString = now.toLocaleDateString() + ' ' + now.toLocaleTimeString();
+
+        // delate old status
+        $draftStatus.empty();
+
+        if(action) {
+            // clone elements and populate
+            var $updateAction = $draftAction.clone().text(action);
+            var $updateTime = $draftTime.clone()
+                .attr('title', now.toISOString())
+                .text(nowString);
+            // append elements
+            $draftStatus.append(gettext('Draft') + ' ')
+                .append($updateAction)
+                .append(' ')
+                .append($updateTime);
```

Comment:
I'm not sure localized strings can be combined this way in all locales.  It may be better to pass the two values in ('published' and 'discarded') and construct translatable strings for each, like
```
if (action === 'published') {
   $draftStatus.append(interpolate("Draft published at %s", $updateTime))
}
```